### ISSUE TITLE
dispose function passed to registerSingleton not called

### ISSUE THREAD

**Issue by mbnadjib** [fluttercommunity/get_it#138]
**GetIt version:** 5.0.1
**Code example:**
```
class Product {
  Product() {
    print('new instance of Product');
  }
}

void main() {
  GetIt.I.registerSingleton<Product>(Product(), dispose: (p) {
    print('$p disposed'); /// <== this is never called
  });
  GetIt.I.unregister<Product>();
}
```
**Behavior:** dispose is not called
**Expected behavior:** dispose should be called on unregister

Am i missing something ?

**Comment by escamoteur** [comment]
I would say you might have found a bug. let me check

**Comment by escamoteur** [comment]
fixed in V5.0.2

**Comment by bogdanmarinescu** [comment]
I'm seeing this problem again in '6.0.0'
